connector_spec/connector_spec.ts

```diff
@@ -0,0 +1,1370 @@
+ * 
+ * KEY DESIGN PRINCIPLE:
+ * The UI should be fully derivable from Zod schemas with minimal additional metadata.
+ * This enables:
+ * - Single source of truth (schema = validation + UI)
+ * - LLM-friendly spec (just define schemas, UI is generated)
+ * - Reduced boilerplate (no separate UI definitions)
+ * 
+ * @see connector_spec.md for comprehensive feature mapping and examples
+ * @see connector_spec_ui.ts for UI metadata extension system
+ * @see connector_spec_examples.ts for example implementations
+ */
+
+import { z } from "zod";
+import type { Logger } from "@kbn/core/server";
+import type { AxiosInstance } from "axios";
+
+// Re-export UI utilities for convenience
+export { withUIMeta, UISchemas } from "./connector_spec_ui";
+
+// ============================================================================
+// METADATA & IDENTIFICATION
+// ============================================================================
+
+/**
+ * Connector metadata that defines its identity and capabilities
+ */
+export interface ConnectorMetadata {
+  /** Unique connector ID (e.g., ".slack_api", ".jira") */
+  id: string;
+  
+  /** Display name shown in UI */
+  displayName: string;
+  
+  /** Icon identifier for UI rendering */
+  icon?: string;
+  
+  /** Short description of the connector's purpose */
+  description: string;
+  
+  /** URL to connector documentation */
+  docsUrl?: string;
+  
+  /** Minimum Kibana license required */
+  minimumLicense: "basic" | "gold" | "platinum" | "enterprise";
+  
+  /** Feature IDs this connector supports */
+  supportedFeatureIds: Array<
+    | "alerting"
+    | "cases" 
+    | "uptime"
+    | "security"
+    | "siem"
+    | "generativeAIForSecurity"
+    | "generativeAIForObservability"
+    | "generativeAIForSearchPlayground"
+    | "endpointSecurity"
+  >;
+  
+  /** 
+   * Sub-feature type for specialized connectors
+   * 
+   * WHY: Some connectors require special categorization beyond basic feature IDs
+   * for security and access control purposes. This allows grouping connectors
+   * that share specialized capabilities or security requirements.
+   * 
+   * USED BY: Endpoint security connectors (Crowdstrike, SentinelOne, Microsoft Defender Endpoint)
+   * REFERENCE: x-pack/platform/plugins/shared/stack_connectors/server/connector_types/crowdstrike/index.ts:42
+   *   Code: `subFeature: 'endpointSecurity'`
+   * REFERENCE: x-pack/platform/plugins/shared/stack_connectors/server/connector_types/sentinelone/index.ts:42
+   *   Code: `subFeature: 'endpointSecurity'`
+   * REFERENCE: x-pack/platform/plugins/shared/stack_connectors/server/connector_types/microsoft_defender_endpoint/index.ts:49
+   *   Code: `subFeature: 'endpointSecurity'`
+   * 
+   * WHY NEEDED: Endpoint security connectors require special privilege checks and 
+   * access controls beyond standard connectors. This flag allows the framework to
+   * apply additional security measures.
+   */
+  subFeature?: "endpointSecurity";
```

Comment:
With us moving the "actions" out of the connectors, I think we can drop this. We added this when we were still putting the actions themselves in the connectors and the actions were "dangerous" and should be used in limited situations.